File: src/java/org/javaturk/dp/ch06/visitor/factory/Director.java
package org.javaturk.dp.ch06.visitor.factory;


public class Director extends Manager {
	protected double bonus;
	
	public Director(int no, String name, int year, String workingDepartment, String managingDepartment, double bonus) {
		super(no, name, year, workingDepartment, managingDepartment);
		this.bonus = bonus;
	}
	
	public void work(){
		System.out.println("Director is working!");
		manage();
	}
	
	public void manage(){
		System.out.println("Director manages whole company!");
		makeAStrategicPlan();
	}
	
	public void makeAStrategicPlan(){
		System.out.println("Director makes a strategic plan for the company!");
	}
	
	public double calculateSalary(){
		return super.calculateSalary() + MANAGEMENT_PAYMENT + bonus;
	}
	
	public void printInfo(){
		System.out.println("\nDirector Info");
		super.printInfo();
	}
}

File: src/c#/Creational/FactoryMethod/FactoryMethod2/Director.cs
using System;
namespace FactoryMethod2
{
    public class Director : Manager
    {
		protected double bonus {get;}

		public Director(int no, String name, int year, String workingDepartment, String managingDepartment, double bonus) : base(no, name, year, workingDepartment, managingDepartment)
		{
			this.bonus = bonus;
		}

		public override void Work()
		{
			Console.WriteLine("Director is working!");
			Manage();
		}

		public override void Manage()
		{
			Console.WriteLine("Director manages whole company!");
			MakeAStrategicPlan();
		}

		public void MakeAStrategicPlan()
		{
			Console.WriteLine("Director makes a strategic plan for the company!");
		}

		public override double CalculateSalary()
		{
			return base.CalculateSalary() + bonus;
		}
	}
}
